src/EnergyPlus/DataPlant.hh

```diff
@@ -597,6 +601,7 @@ namespace DataPlant {
 		int IndexInLoopSidePumps; // If I'm a pump, this tells my index in PL(:)%LS(:)%Pumps
 		Real64 TempDesCondIn;
 		Real64 TempDesEvapOut;
+		std::shared_ptr< PlantComponent > compPtr;
```

Comment:
So I know people have an aversion to 'raw' pointers but what if they were just used for sharing of an object owned by something else? For example,

```cpp
namespace EnergyPlus {
  namespace DataPlant {
    struct CompData
      {
        // Does not own PlantComponent object, just a reference basically
        PlantComponent * compPtr;
      }
  }
  namespace Pipes {
    // This owns the LocalPipeData instances for the lifetime of program 
    // since it is in global namespace.
    Array1D< LocalPipeData > LocalPipe;

    PlantComponent * LocalPipeData::factory( int objectType, std::string objectName ) {
      if ( GetPipeInputFlag ) {
        GetPipeInput();
        GetPipeInputFlag = false;
      }
      for ( auto * pipe : LocalPipe ) {
        if ( pipe->TypeOf == objectType && pipe->name == objectName ) {
          // returns pointer to object stored in Array. 
          // No need to worry about manual memory management since 
          // the Array owns that object and will dispose of it properly.
          return pipe;
        }
      }
      return nullptr;
    }
  }
  namespace PlantManager {
    void
    GetPlantInput()
    {
      // blah blah
      if ( SameString( this_comp_type, "Pipe:Adiabatic" ) ) { // This is awful by the way... sooo many string comparisons
        this_comp.TypeOf_Num = TypeOf_Pipe;
        this_comp.GeneralEquipType = GenEquipTypes_Pipe;
        this_comp.CurOpSchemeType = NoControlOpSchemeType;
        this_comp.compPtr = Pipes::LocalPipeData::factory( TypeOf_Pipe, CompNames( CompNum ) );
      }
      // blah blah
    }
  }
}
```

This is what you are trying to accomplish but without the overhead of shared_ptrs. Now this depends on the Array of objects to have a longer lifetime than any object that has member with a pointer to it.

I dunno, just a thought. Basically we need to figure out ownership and lifetime of objects.

:fire: :see_no_evil: :hear_no_evil: :speak_no_evil: :fire: